### PR TITLE
Add historic curves to the History Matching results metadata

### PR DESCRIPTION
**Pull request by ro-oliveira95**
ASIM-5713

### CODE REVIEW

File: src/alfasim_sdk/result_reader/aggregator.py [hunk 1]
```diff
@@ -300,12 +329,19 @@ def from_result_directory(cls, result_directory: Path) -> Self:
...
+        def map_meta_items(
+            hm_metadata: Dict,
+        ) -> Dict[str, HistoryMatchingMetadata.HMItem]:
             return {
                 key: HistoryMatchingMetadata.HMItem.from_dict(data)
                 for key, data in hm_metadata.items()
             }
 
+        def map_historic_data_infos(
+            infos: list[Dict[str, Any]]
```
Comment (ro-oliveira95) [inline src/alfasim_sdk/result_reader/aggregator.py:341]: - [ ] change to `List`

File: src/alfasim_sdk/result_reader/aggregator.py [hunk 2]
```diff
@@ -1776,6 +1818,30 @@ def read_history_matching_result(
         return result_map
 
 
+def read_history_matching_historic_data_curves(
```
Comment (ro-oliveira95) [inline src/alfasim_sdk/result_reader/aggregator.py:1821]: - [ ] Add test

**Review by nicoddemus — APPROVED** [review]
Do we need a changelog entry?

**Review by prusse-martin — APPROVED** [review]

**Comment by ro-oliveira95** [comment]
@nicoddemus @prusse-martin could you please have a look on the last incoming changes (tests and some restructuring of the `HistoryMatchingMetadata` class) to see if its all good?

**Review by nicoddemus — APPROVED** [review]

**Review by BeneBr — APPROVED** [review]

**Review by prusse-martin — APPROVED** [review]

**Pull request merged**